src/targets/github.ts

```diff
@@ -436,31 +366,12 @@ export class GithubTarget extends BaseTarget {
     if (config.changelogPolicy !== ChangelogPolicy.None) {
       changelog = await this.getChangelog(version);
     }
-    const release = await this.getOrCreateRelease(version, revision, changelog);
 
-    const assets = await this.getAssetsForRelease(release.id);
+    const draft = await this.createReleaseDraft(version, revision, changelog);
+
+    const assets = await this.getAssetsForRelease(draft.id);
     if (assets.length > 0) {
-      this.logger.warn(
-        'Existing assets found for the release, deleting them...'
-      );
-      if (isDryRun()) {
-        this.logger.info('[dry-run] Not deleting assets.');
-      } else {
-        const results = await Promise.allSettled(
-          assets.map(asset => this.deleteAsset(asset))
-        );
-        const failed = results.filter(
-          ({ status }) => status === 'rejected'
-        ) as PromiseRejectedResult[];
-        if (failed.length === 0) {
-          this.logger.debug(`Deleted ${assets.length} assets`);
-        } else {
-          this.logger.debug(
-            'Failed to delete some assets:',
-            failed.map(({ reason }) => reason)
-          );
-        }
-      }
+      throw new Error('Existing assets found for the release draft! Why?');
```

Comment:
Should we have a way to force the deletion/upload despite assets being somehow there in case of an incident?

Reply:
This is really only here because we _definitely_ needed this check for the previous logic where we were creating a true release object each time. In the draft case, each call to `createRelease` starts a new draft, so this is really more of a sanity check than an expected case. Maybe we should remove it? 🤔